extensions/wikia/Wall/VoteHelper.class.php

```diff
@@ -52,162 +47,151 @@ function getVotersList( $from = 0, $count = 50 ) {
-				wfProfileOut( __METHOD__ );
 				return $cacheVal;
 			}
 		}
 
 		$dbr = wfGetDB( $db );
 		// TODO: abstract the query
-		$row = $dbr->selectRow( array( 'page_vote', 'ipblocks' ),
-			array( 'count(*) as cnt' ),
-			array(
+		$row = $dbr->selectRow(
+			[ 'page_vote', 'ipblocks' ],
+			[ 'count(*) as cnt' ],
+			[
 				"ifnull(ipb_expiry, 0) != 'infinity'",
 				'ifnull(ipb_expiry, 0) < ' . wfTimestamp( TS_MW ),
 				'article_id' => $this->pageId,
-			),
+			],
 			__METHOD__,
-			array(
+			[
 				'ORDER BY' => 'time desc'
-			),
-			array( 'ipblocks'  => array( 'LEFT JOIN', 'user_id = ipb_user' ) )
+			],
+			[
+				'ipblocks'  => [ 'LEFT JOIN', 'user_id = ipb_user' ]
+			]
 		);
 
-		$cache->set( $this->getPageCacheKey(), $row->cnt, 60 * 60 );
-		wfProfileOut( __METHOD__ );
+		$cache->set( $this->getPageCacheKey(), $row->cnt );
+
 		return (int) $row->cnt;
 	}
 
 	// TODO: move this to background tasks, for now just invalidation of last 100 votes
 	function invalidateUser() {
-		wfProfileIn( __METHOD__ );
 		$dbr = wfGetDB( DB_SLAVE );
 
-		$res = $dbr->select( array( 'page_vote' ),
-			array( 'article_id' ),
-			array(
-				'user_id' => $this->userId,
-			),
+		$res = $dbr->select(
+			[ 'page_vote' ],
+			[ 'article_id' ],
+			[ 'user_id' => $this->userId ],
 			__METHOD__,
-			array(
+			[
 				'LIMIT' => 100,
 				'ORDER BY' => 'time desc'
-			)
+			]
 		);
 
-		$out = array();
-
 		$cache = $this->getCache();
 		while ( $row = $res->fetchObject() ) {
 			$key = $this->getPageCacheKey( $row->article_id );
 			$cache->delete( $key );
 		}
 
-		wfProfileOut( __METHOD__ );
 		return true;
 	}
 
-	function isVoted( $db = DB_SLAVE ) {
-		wfProfileIn( __METHOD__ );
-		$dbr = wfGetDB( $db );
+	/**
+	 * Whether this user has voted on the forum thread or not.  For anonymous
+	 * users, this always returns false.
+	 *
+	 * @param int $db
+	 *
+	 * @return bool
+	 */
+	public function isVoted( $db = DB_SLAVE ) {
+		if ( $this->userId == 0 ) {
+			return false;
+		}
 
 		$cache = $this->getCache();
-		$cacheVal = $cache->get( $this->getUserCacheKey(), "N" );
+		$cacheVal = $cache->get( $this->getUserCacheKey() );
 
 		if ( !empty( $cacheVal ) ) {
-			wfProfileOut( __METHOD__ );
 			return $cacheVal == 'Y';
 		}
 
-		$row = $dbr->selectRow( 'page_vote',
-			array( 'count(*)as cnt' ),
+		$dbr = wfGetDB( $db );
+		$row = $dbr->selectRow(
+			'page_vote',
+			[ 'count(*) as cnt' ],
 			$this->getUserWhere(),
-			__METHOD__ );
+			__METHOD__
+		);
 
-		$val = ( $row->cnt > 0 );
+		$val = $row->cnt > 0;
 
-		// Y:N solve the problem with defult val
+		// Y:N solve the problem with default val
 		$cache->set( $this->getUserCacheKey(), $val ? 'Y': 'N' );
-		wfProfileOut( __METHOD__ );
 		return $val;
 	}
 
 	protected function getUserWhere() {
-		wfProfileIn( __METHOD__ );
-		$where = array(
-				'article_id' => $this->pageId,
-		);
-
-		if ( $this->userId == 0 ) {
-			$where['user_id'] = 0;
-			$where['ip'] = $this->userIP;
```

Comment:
As I understand this code, it limits number kudos for anon users from one IP address. Was it intended to remove this check?